### PR TITLE
chore: change name of SDK from JavaScript to Node.js

### PR DESCRIPTION
**Pull request by cprice404**

### CODE REVIEW

**Review by schwern — CHANGES_REQUESTED** [review]
Still a lot of references to Javascript.

* README.md
* examples/README.md (and Japanese)
* examples/load-gen.ts
* build-scripts/publish-lambda-layer.sh
* src/grpc/retry-interceptor.ts
* src/internal/momento-signer.ts
* src/simple-cache-client.ts

File: examples/load-gen.ts [hunk 1]
```diff
@@ -364,7 +364,7 @@ ${BasicJavaScriptLoadGen.outputHistogramSummary(loadGenContext.getLatencies)}
 }
 
 const PERFORMANCE_INFORMATION_MESSAGE = `
-Thanks for trying out our basic javascript load generator!  This tool is
+Thanks for trying out our basic node.js load generator!  This tool is
```
Comment (malandis) [inline examples/load-gen.ts:367]:
```suggestion
Thanks for trying out our basic Node.js load generator!  This tool is
```

**Comment by cprice404** [comment]
The 

> > Still a lot of references to Javascript.
> > ```
> > * README.md
> > 
> > * examples/README.md (and Japanese)
> > 
> > * examples/load-gen.ts
> > 
> > * build-scripts/publish-lambda-layer.sh
> > 
> > * src/grpc/retry-interceptor.ts
> > 
> > * src/internal/momento-signer.ts
> > 
> > * src/simple-cache-client.ts
> > ```
> 
> The language is still javascript - it is not my intent to change every occurrence of it.

The README will get updated when this commit is merged because it is generated from a template.

**Comment by cprice404** [comment]
rebased.  also added https://github.com/momentohq/client-sdk-nodejs/pull/235/commits/ca7d439da2c5e0ccc3cf743cb7b7f9fec21e4852 to remove a few more references

cc: @schwern

**Review by schwern — DISMISSED** [review]

**Review by pgautier404 — APPROVED** [review]

**Pull request merged**